mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +57,96 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
                 .setServiceCrawlingOrigin(MicoServiceCrawlingOrigin.GITHUB)
                 .setGitCloneUrl(basicInfoJson.get("clone_url").textValue())
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(uriBasicInfo, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The dockerfile path must be a valid path relativ the the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the github repository. Send to requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a github repository exists
+     * @param uriBasicInfo
+     * @param dockerfilePath
+     * @return
+     */
+    private boolean existsFileInGithubRepo(String uriBasicInfo, String dockerfilePath) {
+        ResponseEntity<String> responseDockerFileInfo;
+        UriComponentsBuilder dockerFileUriBuilder = UriComponentsBuilder.fromHttpUrl(uriBasicInfo);
+        UriComponents dockerFileUriComponent = dockerFileUriBuilder.pathSegment(GITHUB_API_CONTENTS).pathSegment(dockerfilePath).build();
+        log.info("Check if the dockerfile exists at {}",dockerFileUriComponent.toString());
+        responseDockerFileInfo = restTemplate.getForEntity(dockerFileUriComponent.toUri(), String.class);
+        if(!responseDockerFileInfo.getStatusCode().is2xxSuccessful()){
+            log.error("Got an error {} while checking if dockerfile exists",responseDockerFileInfo.getStatusCode());
+            return false;
         }
+        return true;
     }
 
-    public MicoService crawlGitHubRepoLatestRelease(String uri) throws IOException {
+    public MicoService crawlGitHubRepoLatestRelease(String uri, String dockerfilePath) throws IOException {
         uri = adaptUriForGitHubApi(uri);
         String releaseUrl = uri + "/" + RELEASES + "/" + LATEST;
 
-        return crawlGitHubRepo(uri, releaseUrl);
+        return crawlGitHubRepo(uri, releaseUrl, dockerfilePath);
     }
 
-    public MicoService crawlGitHubRepoSpecificRelease(String uri, String version) throws IOException {
+    public MicoService crawlGitHubRepoLatestRelease(String uri) throws IOException {
+        return crawlGitHubRepoLatestRelease(uri,null);
+    }
+
+    public MicoService crawlGitHubRepoSpecificRelease(String uri, String version, String dockerfilePath) throws IOException {
```

Comment:
Consider to annotate the parameter `dockerfilePath` with `@Nullable`.
If you do so, do it also for `crawlGitHubRepo`.

Reply:
Done